src/librbd/cache/pwl/AbstractWriteLog.cc

```diff
@@ -315,7 +318,13 @@ void AbstractWriteLog<I>::log_perf() {
 template <typename I>
 void AbstractWriteLog<I>::periodic_stats() {
   std::lock_guard locker(m_lock);
-  ldout(m_image_ctx.cct, 1) << "STATS: m_log_entries=" << m_log_entries.size()
+  m_cache_state->bytes_allocated = m_bytes_allocated;
+  m_cache_state->bytes_available = m_bytes_allocated_cap - m_bytes_allocated;
+  m_cache_state->bytes_dirty = m_bytes_dirty;
+  m_cache_state->bytes_read = m_perfcounter->get(l_librbd_pwl_rd_bytes);
+  m_cache_state->bytes_read_hit = m_perfcounter->get(l_librbd_pwl_rd_hit_bytes);
+  update_image_cache_state();
```

Comment:
sorry, I don't quite understand that `stats_updated ` timestamp need to added here.  Could you please explain what I should do. As I understand,  `ldout` and `lderr `outputs a timestamp.

Reply:
do you mean that adding timestamp  in `cache state`, when `rbd statue` show `cache state`, it also display when it update?

Reply:
Yes, adding a timestamp field in the cache state blob, so that when the user runs `rbd status`, then can see it and complain if the timestamp is too old (i.e. if the cache state stopped being updated for some reason).